### ISSUE TITLE
Diff syntax highlight doesn't handle all cases

### ISSUE THREAD

**Issue by danilaml** [rouge-ruby/rouge#1022]
For example this diff is not syntax highlighted (but GitHub handles it just fine):

```diff
diff a.txt b.txt 
2d1
< 2
4,5c3
< 4
< 5
---
> 2
```

**Comment by vidarh** [comment]
Thanks. In fact the current rouge diff lexer only handles unified diffs (-u flag). This is the default with git as well, which I presume is why it's not been fixed.

This will fix it; I'll prepare a pull-request as well:

```diff

diff --git adiff --git a/lib/rouge/lexers/diff.rb b/lib/rouge/lexers/diff.rb
index 0f44832..28d2ded 100644
--- a/lib/rouge/lexers/diff.rb
+++ b/lib/rouge/lexers/diff.rb
@@ -19,14 +19,18 @@ module Rouge
 
       state :root do
         rule(/^ .*$\n?/, Text)
-        rule(/^---$\n?/, Text)
+        rule(/^---$\n?/, Punctuation)
+        rule(/^[+>]+.*$\n?/, Generic::Inserted)
         rule(/^\+.*$\n?/, Generic::Inserted)
-        rule(/^-+.*$\n?/, Generic::Deleted)
+        rule(/^[-<]+.*$\n?/, Generic::Deleted)
         rule(/^!.*$\n?/, Generic::Strong)
-        rule(/^@.*$\n?/, Generic::Subheading)
         rule(/^([Ii]ndex|diff).*$\n?/, Generic::Heading)
+        rule(/^(@@[^@]*@@)(.*)$\n?/) do
+          groups Punctuation, Text
+        end
+        rule(/^\w.*$\n?/, Punctuation)
         rule(/^=.*$\n?/, Generic::Heading)
-        rule(/.*$\n?/, Text)
+        rule(/\s.*$\n?/, Text)
       end
     end
   end
```

**Comment by pyrmont** [comment]
This has been fixed by #1068. Thanks @vidarh for getting this one licked :)
